src/script/lua_api/l_mainmenu.cpp

```diff
@@ -862,15 +862,18 @@ int ModApiMainMenu::l_extract_zip(lua_State *L)
 			std::string fullpath = destination;
 			fullpath += DIR_DELIM;
 			fullpath += files_in_zip->getFullFileName(i).c_str();
+			std::string fullpath_dir = fs::RemoveLastPathComponent(fullpath);
 
-			if (files_in_zip->isDirectory(i)) {
-				if (! fs::CreateAllDirs(fullpath) ) {
-					fs->removeFileArchive(fs->getFileArchiveCount()-1);
-					lua_pushboolean(L,false);
-					return 1;
+			if (! files_in_zip->isDirectory(i)) {
+				
+				if (! fs::PathExists(fullpath_dir)) {
```

Comment:
The `!` shouldn't be separated by a space and there shouldn't be an extra newline.
You should also combine the second conditional with the one below.